File: container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        int i = 0, j = height.length-1, ans = 0;
        
        while (i<j){
            int area = (j-i) * Math.min(height[i], height[j]);
            ans = Math.max(ans, area);
            if (height[i]<height[j]) i++;
            else if(height[j]<height[i]) j--;
            else{
                i++;
                j--;
            }
        }
        return ans;
    }
}

File: container-with-most-water/container-with-most-water.py

class Solution:
    def maxArea(self, height: List[int]) -> int:
        """
        186254837
        
        
        """
        i = 0
        j = len(height)-1
        
        ans = 0
        
        while i<j:
            x = j-i
            y = min(height[j], height[i])
            ans = max(x*y, ans)
            if height[i] < height[j]:
                i+=1
            else:
                j-=1
        return ans